libfwupdplugin/fu-common.c

```diff
@@ -3378,3 +3384,102 @@ fu_xmlb_builder_insert_kb (XbBuilderNode *bn, const gchar *key, gboolean value)
 {
 	xb_builder_node_insert_text (bn, key, value ? "true" : "false", NULL);
 }
+
+/**
+ * fu_common_get_firmware_search_path:
+ * @error: (nullable): optional return location for an error
+ *
+ * Reads the FU_PATH_KIND_FIRMWARE_SEARCH and
+ * returns its contents
+ *
+ * Returns: a pointer to a gchar array
+ *
+ * Since: 1.6.2
+ **/
+gchar *
+fu_common_get_firmware_search_path (GError **error)
+{
+	gsize sz = 0;
+	g_autofree gchar *sys_fw_search_path = NULL;
+	g_autofree gchar *contents = NULL;
+
+	sys_fw_search_path = fu_common_get_path (FU_PATH_KIND_FIRMWARE_SEARCH);
+	g_file_get_contents(sys_fw_search_path, &contents, &sz, error);
```

Comment:
if this returns FALSE, return FALSE. Also space before the bracket.

Reply:
fixed